### DISCUSSION TITLE
Marks have `begin` and `end` - maybe `begin` and `length` would be better

### DISCUSSION THREAD

**Discussion by orionz** [automerge/automerge#1518] [General]
@jkankiewicz pointed out that `end < begin` is an un needed error case - if we we instead passed in an unsigned length we'd have one less error case to handle.  Is making an API change worth it? Thoughts?

**Comment by heckj** [comment]
For what it's worth, length would be a smidge easier to compute than an end index when working with the Swift language overlay. Either is workable, but length seems more consistent with the rest of the API here.

**Comment by alexjg** [comment]
Begin and length makes sense to me.
